syntax = "proto3";

package blockchain;

message Block {
    message Header {
        uint32 version = 1;
        bytes merkle_root = 2;
        bytes parent_block = 3;
        uint32 timestamp = 4;
    }

    message TreeBranchData {
        bytes parent_hash = 1;
        bytes data = 2;
    }

    Header header = 1;
    repeated TreeBranchData branch_data = 2;
}
